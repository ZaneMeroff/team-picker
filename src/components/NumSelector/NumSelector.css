.picker-outer-container {
  margin: 10px;
}

.picker-num-btn-container {
  border: 2px solid #000;
  height: 100px;
  width: 150px;
  display: flex;
}

.plus-minus-button-container {
  display: flex;
  flex-direction: column;
  width: 50%;
}

.number-display-container {
  width: 50%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.number-display {
  font-size: 300%;
}

.increment-button {
  height: 50%;
  border-radius: 0px;
  border: 0px;
  border-left: 2px solid #000;
  font-size: 200%;
}

.increment-button:hover {
  cursor: pointer;
  background-color: yellow;
}
