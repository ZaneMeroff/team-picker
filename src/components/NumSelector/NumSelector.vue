<template>
  <div>
    <p>pick a number of teams:</p>
    <p>0</p>
    <button>+</button>
    <button>-</button>
  </div>
</template>

<script>
export default {
  name: 'NumSelector',
  props: {

  }
}
</script>

<style>

</style>
