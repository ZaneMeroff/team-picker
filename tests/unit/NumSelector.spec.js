import { shallowMount } from '@vue/test-utils'
import NumSelector from '../../src/components/NumSelector/NumSelector.vue'

describe('NumSelector.vue', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(NumSelector, {
      propsData: {
        label: 'teams',
        number: 2,
        updateNumber: jest.fn()
      }
    })
  })

  it('component render matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('component has correct props', () => {
    expect(wrapper.props().label).toEqual('teams')
    expect(wrapper.props().number).toEqual(2)
    expect(typeof wrapper.props().updateNumber).toEqual('function')
  })

  it('component has correct default data values', () => {
    expect(wrapper.vm.$data.buttonTypeUp).toEqual('teamsUp')
    expect(wrapper.vm.$data.buttonTypeDown).toEqual('teamsDown')
  })

})
