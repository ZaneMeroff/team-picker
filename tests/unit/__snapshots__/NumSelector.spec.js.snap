// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumSelector.vue component render matches snapshot 1`] = `
<section class="picker-outer-container">
  <p>number of teams:</p>
  <div class="picker-num-btn-container">
    <div class="number-display-container">
      <p class="number-display">2</p>
    </div>
    <div class="plus-minus-button-container"><button class="increment-button-up">⬆</button> <button class="increment-button-down">⬇</button></div>
  </div>
</section>
`;
